lightning/src/ln/outbound_payment.rs

```diff
@@ -849,14 +856,18 @@ impl OutboundPayments {
 		match self.pending_outbound_payments.lock().unwrap().entry(payment_id) {
 			hash_map::Entry::Occupied(entry) => match entry.get() {
 				PendingOutboundPayment::AwaitingInvoice {
-					retry_strategy: retry, max_total_routing_fee_msat: max_total_fee, ..
+					retry_strategy: retry, max_total_routing_fee_msat: max_total_fee, route_params_config, ..
 				} => {
 					retry_strategy = *retry;
-					max_total_routing_fee_msat = *max_total_fee;
+					// If max_total_fee is present, update route_params_config with the specified fee.
+					// This supports the standard behavior during downgrades.					
```

Comment:
That's the behavior needed when upgrading, not downgrading. For downgrades, we just need to make sure we continue to write `max_total_routing_fee_msat`.

Reply:
Makes sense! Looks like I mixed up supporting downgrade with upgrade :)
Thanks for the pointer!
Fixed in [pr3342.06](https://github.com/shaavan/rust-lightning/commits/pr3342.06)